### PR TITLE
Fix StreamGear Malformed URI Error with Segments

### LINKED ISSUE #248: StreamGear Malformed URI with Segments
## Description

- I am using StreamGear Real-Time Frames mode to transmit a live feed from a webcam to a client.
- I am using StreamGear to generate the playlist file (m3u8) and the segments (.ts files).
- When using the StreamGear Real-Time Frames mode, the segment files contain a '%' character in their name which causes an error when I am serving this static content through Nginx.
- The playlist file is served without an issue.

### Acknowledgment

- [x] I have searched the [issues](https://github.com/abhiTronix/vidgear/issues) for my issue and found nothing related or helpful.
- [x] I have read the [Documentation](https://abhitronix.github.io/vidgear/latest).
- [x] I have read the [Issue Guidelines](https://abhitronix.github.io/vidgear/latest/contribution/issue/#submitting-an-issue-guidelines).

### Environment

* VidGear version:   0.2.2
* Branch:  Master
* Python version:  3.6.9
* PiP version:  21.2.4
* Operating System and version: Ubuntu 18.04.6 LTS (Arm64)

### Expected Behavior

The segment files should be served without an issue by a server such as Nginx.

### Actual Behavior

The segment files are not served by Nginx and they result in a malformed URI error

### Possible Fix

Removing the extra '%' character from the naming convention for segment files

### Steps to reproduce

(Write your steps here:)

1. Create a StreamGear object with the following parameters:
- format="hls"
- stream_params: {"-livstream": True and "-input_framerate": 25}
- output="/test/folder"
2. Serve the folder /test/folder as static content with Nginx
3. Try to access the live stream through a javascript video player such as video.js and look at the network tab in your browser

### Example of the Bug:
Screenshot-20210930011311-1747x487
Screenshot-20210930011436-644x36

### PR DESCRIPTION
**Pull request by Vboivin**
## Description
- I removed the extra '%' character from the naming convention for segment files.
- I removed the Windows carriage returns from the shell scripts to be able to execute them on Linux.
### Requirements / Checklist

- [x] I have read the [PR Guidelines](https://abhitronix.github.io/vidgear/latest/contribution/PR/#submitting-pull-requestpr-guidelines).
- [x] I have read the [Documentation](https://abhitronix.github.io/vidgear/latest).
- [x] I have updated the documentation accordingly(if required).

### Related Issue

Closes #248

### Context

The segment files are now served without an issue by a server such as Nginx.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Before Fix
Screenshot-20210930011311-1747x487
Screenshot-20210930011436-644x36
#### After Fix
Screenshot-20210930011757-1739x403

### CODE REVIEW

**Comment by Vboivin** [comment]
@abhiTronix I modified the code to use stream_count to define which segment_template we want to use for the filenames.

File: vidgear/gears/streamgear.py [hunk 1]
```diff
@@ -810,7 +810,10 @@ def __generate_hls_stream(self, input_params, output_params):
 
         # Finally, some hardcoded HLS parameters (Refer FFmpeg docs for more info.)
         output_params["-allowed_extensions"] = "ALL"
-        output_params["-hls_segment_filename"] = "{}-stream-%03d.{}".format(
+        # Normally output_params.get("stream_count", 0) would be used to access a dict variable,
+        # but in this case stream_count is always defined by default with 0
+        segment_template = "{}-stream%v-%03d.{}" if output_params["stream_count"] > 1 else "{}-stream-%03d.{}"
```
Comment (abhiTronix) [inline vidgear/gears/streamgear.py:815]: @Vboivin  logic is wrong, `output_params["stream_count"] > 1` must be `output_params["stream_count"] > 0` instead.

**Comment by abhiTronix** [comment]
> @abhiTronix I modified the code to use stream_count to define which segment_template we want to use for the filenames.

@Vboivin Logic is still wrong, please see my review.

**Review by abhiTronix — APPROVED** [review]
Nevermind you're correct, `output_params["stream_count"]` value must be greater than `1` and not `0` to detect multi-variant stream.

**Comment by abhiTronix** [comment]
@Vboivin Thanks! Once CI tests are completed successfully, I'll merge it. 👍

**Pull request merged**